LadybugTools_Engine/Create/AnalysisPeriod.cs

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base.Attributes;
+using BH.oM.LadybugTools;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace BH.Engine.LadybugTools
+{
+    public static partial class Create
+    {
+        [Description("Create an AnalysisPeriod object.")]
+        [Input("stMonth", "The starting month of the analysis period. 1-12")]
+        [Input("stDay", "The starting day of the analysis period. 1-31")]
+        [Input("stHour", "The starting hour of the analysis period. 0-23")]
+        [Input("endMonth", "The ending month of the analysis period. 1-12")]
+        [Input("endDay", "The ending day of the analysis period. 1-31")]
+        [Input("endHour", "The ending hour of the analysis period. 0-23")]
+        [Input("isLeapYear", "Whether the analysis period represents a leap year.")]
+        [Input("timestep", "The number of timesteps per hour. One of {1, 2, 3, 4, 5, 6, 10, 12, 15, 20, 30, 60}")]
+        [Output("analysisPeriod", "An AnalysisPeriod object.")]
+        public static AnalysisPeriod AnalysisPeriod(int stMonth = 1, int stDay = 1, int stHour = 0, int endMonth = 12, int endDay = 31, int endHour = 23, bool isLeapYear = false, int timestep = 1)
```

Comment:
seeing as these variable names aren't used by the serialiser, could they be clearer? (eg. `stMonth` to `startMonth`)

Reply:
Seconded 👍

Reply:
But they are used during serialisation as the Python-side is expecting an AnalysisPeriod dict with key names per those given on the objects currently. If we hide the functionality within an adapter then I'm happy to change these on BHoM-side and have it rename upon conversion to a JSON-string ... but for now it's as simple as `obj.ToJson()` without any additional code required.

Reply:
The `AnalysisPeriod` object itself can stay the same, as the object is what is used for serialisation. The serialiser doesn't see this method - rather the object itself - so variable names in this method can be anything as long as the object stays the same.

Reply:
> The `AnalysisPeriod` object itself can stay the same, as the object is what is used for serialisation. The serialiser doesn't see this method - rather the object itself - so variable names in this method can be anything as long as the object stays the same.

Good point! I'll make that change to this method.